doozer/doozerlib/backend/rebaser.py

```
@@ -1921,17 +1933,20 @@ async def _resolve_image_from_upstream_parent(self, original_parent: str, dfp: D
 
         try:
             self._logger.debug('Retrieving image info for image %s', original_parent)
-            labels = await util.oc_image_info_for_arch_async__caching(original_parent)['config']['config']['Labels']
 
-            # Get builder X.Y
-            major, minor, _ = util.extract_version_fields(labels['version'])
+            # Use the cached function to extract builder info
+            el_version, golang_version = extract_builder_info_from_pullspec(original_parent)
+
+            if not el_version or not golang_version:
+                raise ValueError(f'Could not extract RHEL version or golang version from {original_parent}')
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)